src/main/java/io/snyk/jenkins/SnykStepFlow.java

```diff
@@ -17,53 +17,57 @@
 
 public class SnykStepFlow {
 
-  public static void perform(SnykConfig config, Supplier<SnykContext> contextSupplier) throws SnykIssueException, SnykErrorException {
-    int testExitCode = 0;
+  public static void perform(SnykConfig config, Supplier<SnykContext> contextSupplier)
+  throws SnykIssueException, SnykErrorException {
+    boolean foundIssues = false;
     Exception cause = null;
     SnykContext context = null;
 
     try {
       context = contextSupplier.get();
-      testExitCode = SnykStepFlow.scan(context, config);
+
+      SnykInstallation installation = SnykInstallation.install(
+        context,
+        config.getSnykInstallation()
+      );
+
+      context.getEnvVars().put("SNYK_TOKEN", SnykApiToken.getToken(context, config.getSnykTokenId()));
+
+      foundIssues = SnykStepFlow.testProject(context, config, installation);
+
+      if (config.isMonitorProjectOnBuild()) {
+        SnykMonitor.monitorProject(context, config, installation);
+      }
```

Comment:
I moved the monitor step to be after the HTML step as the SnykTest and SnykToHTML steps rely on each other.